fe/fe-core/src/main/java/com/starrocks/service/arrow/flight/sql/session/ArrowFlightSqlSessionManager.java

```diff
@@ -17,59 +17,117 @@
+                    "Maybe the token is expired or evicted, could modify fe.conf " +
+                    "[arrow_token_cache_expire_second] and [arrow_token_cache_size]", token));
+        }
     }
 
-    public ArrowFlightSqlConnectContext getConnectContext(String peerIdentity) {
-        try {
-            ArrowFlightSqlTokenInfo arrowFlightSqlTokenInfo =
-                    arrowFlightSqlTokenManager.validateToken(peerIdentity);
+    public void closeSession(String token) {
+        tokenCache.invalidate(token);
+    }
 
-            String token = arrowFlightSqlTokenInfo.getToken();
-            ArrowFlightSqlConnectContext arrowFlightSqlConnectContext =
-                    ExecuteEnv.getInstance().getScheduler().getArrowFlightSqlConnectContext(token);
-            if (arrowFlightSqlConnectContext == null) {
-                arrowFlightSqlConnectContext =
-                        createArrowFlightSqlConnectContext(token, arrowFlightSqlTokenInfo);
-            }
-            return arrowFlightSqlConnectContext;
-        } catch (Exception e) {
-            throw CallStatus.INTERNAL.withCause(e).toRuntimeException();
+    @NotNull
+    public ArrowFlightSqlConnectContext validateAndGetConnectContext(String token) throws FlightRuntimeException {
+        ArrowFlightSqlConnectContext connectContext =
+                ExecuteEnv.getInstance().getScheduler().getArrowFlightSqlConnectContext(token);
+        if (connectContext == null) {
+            throw CallStatus.NOT_FOUND
+                    .withDescription("cannot find connect arrow context of the token [" + token + "]")
+                    .toRuntimeException();
         }
+        return connectContext;
+    }
+
+    private ArrowFlightSqlTokenInfo createToken(UserIdentity currentUser) {
+        String token = UUIDUtil.genUUID().toString();
+        ArrowFlightSqlTokenInfo tokenInfo = new ArrowFlightSqlTokenInfo(currentUser, token);
+        tokenCache.put(token, tokenInfo);
+        return tokenInfo;
     }
 
-    private ArrowFlightSqlConnectContext createArrowFlightSqlConnectContext(String token,
-                                                                            ArrowFlightSqlTokenInfo arrowFlightSqlTokenInfo) {
-        UserIdentity currentUser = arrowFlightSqlTokenInfo.getCurrentUser();
-        ArrowFlightSqlConnectContext ctx = new ArrowFlightSqlConnectContext();
+    private void createConnectContext(ArrowFlightSqlTokenInfo tokenInfo) throws FlightRuntimeException {
+        UserIdentity currentUser = tokenInfo.getCurrentUser();
+        ArrowFlightSqlConnectContext ctx = new ArrowFlightSqlConnectContext(tokenInfo.getToken());
+
         ctx.setGlobalStateMgr(GlobalStateMgr.getCurrentState());
-        ctx.setQualifiedUser(currentUser.getUser());
         ctx.setQueryId(UUIDUtil.genUUID());
+        ctx.setExecutionId(UUIDUtil.toTUniqueId(ctx.getQueryId()));
```

Comment:
This part of the logic already exists in AuthenticationHandler::setAuthenticationResultToContext and should not be set separately, because this will cause logical inconsistency or missing functions. So I think the ConnectionContext should be created before AuthenticationHandler.authenticate.